server/src/main/java/org/opensearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -332,29 +333,57 @@ public void collect(int doc, long owningBucketOrd) throws IOException {
         });
     }
 
+    @Override
+    public List<String> getDimensionFilters() {
+        List<String> dimensionsToMerge = new ArrayList<>();
+        dimensionsToMerge.add(fieldName);
+
+        for (Aggregator subAgg : subAggregators) {
+            if (subAgg instanceof StarTreePreComputeCollector collector) {
+                List<String> childFilters = collector.getDimensionFilters();
+                dimensionsToMerge.addAll(childFilters != null ? childFilters : Collections.emptyList());
+            }
+        }
+
+        return dimensionsToMerge;
+    }
+
     public StarTreeBucketCollector getStarTreeBucketCollector(
         LeafReaderContext ctx,
         CompositeIndexFieldInfo starTree,
         StarTreeBucketCollector parent
     ) throws IOException {
-        assert parent == null;
         StarTreeValues starTreeValues = StarTreeQueryHelper.getStarTreeValues(ctx, starTree);
         SortedSetStarTreeValuesIterator valuesIterator = (SortedSetStarTreeValuesIterator) starTreeValues.getDimensionValuesIterator(
             fieldName
         );
         SortedNumericStarTreeValuesIterator docCountsIterator = StarTreeQueryHelper.getDocCountsIterator(starTreeValues, starTree);
+        List<String> dimensionsToMerge = getDimensionFilters();
+
+        /* For nested aggregations, we require the RemapGlobalOrdsStarTree strategy to properly
+        handle global ordinal remapping. This check ensures we don't reinitialize the
+        collectionStrategy again if it's already correctly set. */
+        if (parent != null && !(collectionStrategy instanceof RemapGlobalOrdsStarTree)) {
```

Comment:
I had fixed redundant initialization of globalOrds as part of this change https://github.com/opensearch-project/OpenSearch/pull/18059 (pending merge)

Make sure that the changes don't break.